compiler/rustc_hir_analysis/src/collect/generics_of.rs

```diff
@@ -58,6 +58,13 @@ pub(super) fn generics_of(tcx: TyCtxt<'_>, def_id: LocalDefId) -> ty::Generics {
     let hir_id = tcx.local_def_id_to_hir_id(def_id);
 
     let node = tcx.hir_node(hir_id);
+
+    if let Some(sig) = node.fn_sig()
+        && let Some(sig_id) = sig.decl.opt_delegation_sig_id()
+    {
+        return inherit_generics_for_delegation_item(tcx, def_id, sig_id);
+    }
```

Comment:
I assume that regression was caused by `opt_hir_owner_nodes` call.  But `OwnerNode` is already extracted here. It would be better for readability to  have 1 branch here instead of 3 by `OwnerNode::Item`,
`OwnerNode::TraitItem`, `OwnerNode::ImplItem`. Maybe run perf?

Reply:
I'll run perf after the PR is otherwise completed.